server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -461,7 +461,6 @@ private static void innerParseObject(DocumentParserContext context, ObjectMapper
         while (token != XContentParser.Token.END_OBJECT) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = context.parser().currentName();
-                splitAndValidatePath(currentFieldName);
```

Comment:
is it ok to omit this call now? What was it doing and where is that method called now? Seems like the new method incorporates some of its logic?

Reply:
The new method is pretty much a straight copy.  It's called in the DotExpandingXContentParser, which InternalDocumentParserContext creates to wrap the source document's xcontent parser.  The existing call sites were places where we previously split up field names, but that all happens in the expanding parser now so it makes more sense to handle things there.

Reply:
ok then I guess I am unclear on why the original method stays around, are there still places where we call it?

Reply:
yep it's still called from `createDynamicUpdate()`.  This gets completely refactored in #81449 so it will go away entirely there.

Reply:
I see, thanks! Can we in the meantime share the code between the two then, or are there subtle differences?

Reply:
They are slightly different because the x-content lib doesn't have access to `Strings`, but in implementation they are the same.  Given that they're `private static` though, and that the copy in `DocumentParser` will be removed immediately in a follow up, I'd prefer to keep two versions for the moment?